js/components/Star/index.js

```diff
@@ -0,0 +1,48 @@
+import React, { Component, PropTypes } from 'react';
+import {
+  TouchableOpacity,
+  Image,
+  StyleSheet,
+} from 'react-native';
+
+export default class Star extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      value: false,
+    };
+
+    this.toggle = this.toggle.bind(this);
+  }
+
+  toggle() {
+    this.setState({ value: !this.state.value });
+  }
+
+  render() {
+    const img = this.state.value ?
+      <Image source={require('./assets/star.png')}/> :
+      <Image source={require('./assets/star-border.png')}/>;
+
+    return (
+      <TouchableOpacity
+        style={styles.container}
+        onPress={this.toggle}>
```

Comment:
Should we add a callback property to inform the user of this component that the toggle happened therefore an action following that 'starred' can happened?

Or do you prefer the opposite logic, perform an action in the parent component and when that action is finished change the property used to show the start?